cmake/cargo.cmake

```diff
@@ -157,6 +157,10 @@ function(add_crate crate)
     list(APPEND build_args "$<IF:$<CONFIG:Debug>,--profile=dev,--release>")
     list(APPEND build_args ${arg_ARGS})
 
+    if(arg_LIBRARY)
+        list(APPEND build_args "--lib")
+    endif()
+
```

Comment:
This one should be inside `staticlib` block below so it works with a single type crate.